### ISSUE TITLE
Wrong tooltip `Dictionary key` when indexing a list

### ISSUE THREAD

**Issue by Viicos** [microsoft/pylance-release#4689]
## Environment data

-   Language Server version: 2023.7.40 (pyright b917b7a5)
-   OS and version: Ubuntu
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10

## Code Snippet

```python
def func(lst: list[str]):
    if len(lst) >= 2:
        return (lst[0], lst[1])
    return lst[1]
```

## Repro Steps

In some cases, pylance shows a tooltip when using indexes on a list (when it knows it is of a certain length). The tooltip, however, shows the message `Dictionary key`.

## Expected behavior

Should show a tooltip saying `List key` or similar

## Actual behavior

![image](https://github.com/microsoft/pylance-release/assets/65306057/afd76f5c-e49c-44e5-b162-5a24b7882639)

## Logs

```
XXX
```

**Comment by erictraut** [comment]
Yeah, the text is incorrect. The completion provider supports completions for any class that supports `__getitem__` which includes `list`, `Sequence`, etc. It doesn't need to be a `dict`. The text "Dictionary key" should probably be replaced by something more general like "Index value". (I'll note that this text is also not currently localized. If you're going to change this text, it might be worth making it localizable at the same time.)

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.9.11, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023911-6-September-2023-PreRelease)
